pkg/storage/labels/labels.go

```diff
@@ -20,6 +21,9 @@ func New(db *badger.DB) *Labels {
 func (ll *Labels) PutLabels(labels map[string]string) error {
 	return ll.db.Update(func(txn *badger.Txn) error {
 		for k, v := range labels {
+			if err := flameql.ValidateTagKeyWithoutReserved(k); err != nil {
+				return err
+			}
```

Comment:
I like that now labels are validated here, but I'm afraid when we call `storage.Put` we need to validate them earlier, before the key is accessed for the first time in this section:
https://github.com/pyroscope-io/pyroscope/blob/65e1d6923461cb29ca46fe777bd6355bbaba1d06/pkg/storage/storage_put.go#L37-L66

Otherwise, we may store invalid app name or invalid labels / segment key (if it is an exemplar)